src/state/CalculationLogics.ts

```diff
@@ -0,0 +1,219 @@
+import i18n from "i18next";
+import { Cycle } from "../data/ClassCycle";
+
+const millisecondsInDay = 24 * 60 * 60 * 1000;
+
+export function getOvulationStatus(cycleLength: number, dayOfCycle: number) {
+  if (!cycleLength || !dayOfCycle) {
+    return "";
+  }
+
+  const lutealPhaseLength = 14;
+  const ovulationDay = cycleLength - lutealPhaseLength;
+  const diffDay = ovulationDay - dayOfCycle;
+  if (diffDay === 0) {
+    return i18n.t("today");
+  } else if (diffDay < 0 && diffDay >= -2) {
+    return i18n.t("possible");
+  } else if (diffDay < 0) {
+    return i18n.t("finished");
+  } else if (diffDay === 1) {
+    return i18n.t("tomorrow");
+  } else if (diffDay < 5) {
+    return `${i18n.t("in")} ${diffDay} ${i18n.t("Days less 5")}`;
+  }
+  return `${i18n.t("in")} ${diffDay} ${i18n.t("Days")}`;
+}
+
+export function getPregnancyChance(cycleLength: number, dayOfCycle: number) {
+  if (!cycleLength || !dayOfCycle) {
+    return "";
+  }
+
+  const lutealPhaseLength = 14;
+  const ovulationDay = cycleLength - lutealPhaseLength;
+  const diffDay = ovulationDay - dayOfCycle;
+
+  if (diffDay <= 4 && diffDay >= -2) {
+    return i18n.t("high");
+  }
+  return i18n.t("low");
+}
+
+export function getDaysBeforePeriod(cycleLength: number, startDate: string) {
+  if (!startDate || !cycleLength) {
+    return {
+      title: i18n.t("Period in"),
+      days: i18n.t("no info"),
+    };
+  }
+
+  const dateOfFinish = new Date(startDate);
+  dateOfFinish.setDate(dateOfFinish.getDate() + cycleLength);
+  dateOfFinish.setHours(0, 0, 0, 0);
+  const now = new Date();
+  now.setHours(0, 0, 0, 0);
+  const dayBefore = Math.round(
+    (Number(dateOfFinish) - Number(now)) / millisecondsInDay,
+  );
+
+  if (dayBefore > 0) {
+    return {
+      title: i18n.t("Period in"),
+      days: `${dayBefore} ${i18n.t("Days_interval", {
```

Comment:
Why here without Math.abs?

Reply:
In this situation, ```dayBefore``` is a positive number. Math.abs is not needed here.